### PR TITLE
Add GCS Fuse Storage to workload creation

### PR DESCRIPTION
**Pull request by PBundyra**
## Fixes / Features
- Add PersistentVolume and PersistentVolumeClaims to the Pods definitions so workload can use GCS Fuse Storage

## Testing / Documentation
Testing details.

- [ y ] Tests pass
- [ y ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: src/xpk/core/storage.py [hunk 1]
```diff
@@ -257,6 +258,131 @@ def install_storage_crd(k8s_api_client: ApiClient) -> None:
...
+  each Storage.
+
+  Args:
+      storages: A list of Storage objects.
+
+  Returns:
+      A string containing the YAML representation of the volumes section.
+  """
+  yaml_str = ""
+  for storage in storages:
```
Comment (pawloch00) [inline src/xpk/core/storage.py:344]: if there are multiple storages shouldn't we append \n on end of each persistentVolumeClaim?

File: src/xpk/core/core.py [hunk 2]
```diff
@@ -2609,15 +2636,23 @@ def get_volume_mounts(args, system: SystemCharacteristics) -> str:
...
+                """
     elif system.device_type == h100_mega_device_type:
       volume_mount_yaml = """- name: nvidia-install-dir-host
                   mountPath: /usr/local/nvidia/lib64
                 - name: shared-memory
                   mountPath: /dev/shm
                 - name: workload-terminated-volume
-                  mountPath: /usr/share/workload"""
-
+                  mountPath: /usr/share/workload
```
Comment (pawloch00) [inline src/xpk/core/core.py:2647]: Can /usr/share/workload and other value sof mountPath in yamls be exported to variables?

Comment (PBundyra) [reply src/xpk/core/core.py:2647]: I like the idea, but I believe it's out of the scope of this PR

File: src/xpk/core/storage.py [hunk 3]
```diff
@@ -257,6 +258,131 @@ def install_storage_crd(k8s_api_client: ApiClient) -> None:
...
+    yaml_str += f"""- name: {storage.pv}
+              persistentVolumeClaim:
+                claimName: {storage.pvc}
+                readOnly: {storage.readonly}
+            """
+  return yaml_str
+
+
+def get_storage_volume_mounts_yaml_for_gpu(storages: list[Storage]) -> str:
```
Comment (44past4) [inline src/xpk/core/storage.py:307]: Why we need this? It seems to be a copy paste of get_storage_volume_mounts_yaml.

File: src/xpk/core/storage.py [hunk 4]
```diff
@@ -257,6 +258,131 @@ def install_storage_crd(k8s_api_client: ApiClient) -> None:
...
+  yaml_str = ""
+  for storage in storages:
+    yaml_str += f"""- name: {storage.pv}
+                  mountPath: {storage.mount_point}
+                  readOnly: {storage.readonly}
+            """
+  return yaml_str
+
+
+def get_storage_volumes_yaml_for_gpu(storages: list[Storage]) -> str:
```
Comment (44past4) [inline src/xpk/core/storage.py:329]: Why we need this? It seems to be a copy paste of get_storage_volumes_yaml.

File: src/xpk/core/storage.py [hunk 5]
```diff
@@ -257,6 +258,131 @@ def install_storage_crd(k8s_api_client: ApiClient) -> None:
...
+
+    member = (
+        f"principal://iam.googleapis.com/projects/{args.project_number}/"
+        f"locations/global/workloadIdentityPools/{args.project}.svc.id.goog/"
+        f"subject/ns/default/sa/{XPK_SA}"
+    )
+
+    policy.bindings.append({"role": role, "members": {member}})
+    bucket.set_iam_policy(policy)
+    print(f"Added {member} with role {role} to {storage.bucket}.")
```
Comment (44past4) [inline src/xpk/core/storage.py:383]: Why not xpk_print?

File: src/xpk/commands/workload.py [hunk 6]
```diff
@@ -402,6 +434,16 @@ def workload_create(args) -> None:
     if return_code != 0:
       xpk_exit(return_code)
 
+  storages: list[Storage] = get_storages(k8s_api_client, args.storage)
+  gcs_fuse_annotation = ''
+  service_account = ''
+  if len(storages) > 0:
+    gcs_fuse_annotation = GCS_FUSE_ANNOTATION
```
Comment (44past4) [inline src/xpk/commands/workload.py:441]: Should we check types of the storages?

File: src/xpk/commands/workload.py [hunk 7]
```diff
@@ -423,6 +465,10 @@ def workload_create(args) -> None:
         gpu_rxdm_image=get_gpu_rxdm_image(system),
         gpu_rxdm_cmd=get_gpu_rxdm_cmd(system),
         gpu_tcp_volume=get_gpu_tcp_volume(system),
+        storage_volumes=get_storage_volumes_yaml_for_gpu(storages),
+        storage_volume_mounts=get_storage_volume_mounts_yaml_for_gpu(storages),
+        gcs_fuse_annotation=gcs_fuse_annotation,
+        service_account=XPK_SA,
```
Comment (44past4) [inline src/xpk/commands/workload.py:471]: Why is it has a fixed value of XPK_SA and below we use service_account as a value?

File: src/xpk/commands/workload.py [hunk 8]
```diff
@@ -402,6 +434,16 @@ def workload_create(args) -> None:
     if return_code != 0:
       xpk_exit(return_code)
 
+  storages: list[Storage] = get_storages(k8s_api_client, args.storage)
+  gcs_fuse_annotation = ''
```
Comment (44past4) [inline src/xpk/commands/workload.py:438]: I would change the name to something more generic like storage_annotations and change the name in the templates above as well.

File: src/xpk/core/storage.py (hunk 3 above)
Comment (PBundyra) [reply src/xpk/core/storage.py:307]: The indentation is different

File: src/xpk/core/storage.py (hunk 4 above)
Comment (PBundyra) [reply src/xpk/core/storage.py:329]: The identation is different

File: src/xpk/core/storage.py (hunk 5 above)
Comment (PBundyra) [reply src/xpk/core/storage.py]: Fixed

File: src/xpk/commands/workload.py (hunk 6 above)
Comment (PBundyra) [reply src/xpk/commands/workload.py]: Done

File: src/xpk/commands/workload.py (hunk 7 above)
Comment (PBundyra) [reply src/xpk/commands/workload.py]: Fixed

File: src/xpk/commands/workload.py (hunk 8 above)
Comment (PBundyra) [reply src/xpk/commands/workload.py]: Done

**Review by pawloch00 — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -330,18 +330,37 @@ Once it's ready user can define
...
+`--manifest` - defines the path to manifest which contains PersistentVolue and PersistentVolumeClaim definitions
 
 
-* Create simple Storage
+* Create a simple Storage
 
     ```shell
     python3 xpk.py storage create test-storage --project=$PROJECT
-    --cluster=xpk-test --type=test-type --auto-mount=false \
+    --cluster=xpk-test --type=gcs-fuse --auto-mount=false \
```
Comment (44past4) [inline README.md:340]: I think that instead of gcs-fuse this should be gcsfuse.

Comment (PBundyra) [reply README.md:340]: Done

**Review by 44past4 — APPROVED** [review]

**Pull request merged**